### DISCUSSION TITLE
API - Getting the reason that a pull request isn't mergeable?

### DISCUSSION THREAD

**Discussion by bfops** [community/community#21886] [New to GitHub]
<p>I can use the <a href="https://developer.github.com/v3/pulls/" rel="nofollow noopener">pull requests</a> api to see whether a PR is mergeable, and I can use the <a href="https://developer.github.com/v3/repos/statuses/" rel="nofollow noopener">status api</a> to check custom statuses logged for a given commit, but how can I fetch the reason that a PR is blocked from merging (e.g. it could be merge conflicts, review obligations, commit status)?</p>

**Comment by lee-dohm** [comment]
<p>There isn’t an API that summarizes why a PR is mergeable or not, only that it is or isn’t. In <a href="https://developer.github.com/v3/pulls/#get-a-single-pull-request" rel="nofollow noopener">the documentation</a> it states:</p>
<blockquote>
<p>The value of the <code>mergeable</code> attribute can be <code>true</code>, <code>false</code>, or <code>null</code>. If the value is <code>null</code>, this means that the mergeability hasn’t been computed yet, and a background job was started to compute it. Give the job a few moments to complete, and then submit the request again. When the job is complete, the response will include a non-<code>null</code> value for the <code>mergeable</code> attribute.</p>
</blockquote>
<p>We don’t record the result of why, only yes, no, or not yet computed.</p>
<p>I hope that helps :+1:</p>

**Comment by bfops** [comment, marked as answer]
<p>I also found the undocumented attribute <code>mergeable_state</code>, which gives a little more info: it can be <code>clean</code>, <code>dirty</code>, <code>blocked</code>, <code>unstable</code>, or <code>unknown</code>. It seems like <code>dirty</code> means merge conflicts, so that’s been granular enough for me for now.</p>
<p>Thanks!</p>
